picard/tagger.py

```diff
@@ -511,81 +544,117 @@ def add_files(self, filenames, target=None):
+
+    def _file_loaded_thread(self):
+        while True:
+            file, target = self._file_loaded_queue.get()
+
+            if self.stopping:
+                break
+
+            thread.to_main(self._file_loaded, file, target)
+            if not self._pending_files_results_queue.empty():
+                time.sleep(5)
+            elif not self._pending_load_finished_queue.empty():
+                time.sleep(2)
             else:
-                number_of_files = len(files)
-                if number_of_files:
-                    mparms = {
-                        'count': number_of_files,
-                        'directory': root,
-                    }
-                    log.debug("Adding %(count)d files from '%(directory)r'" %
-                              mparms)
-                    self.window.set_statusbar_message(
-                        ngettext(
-                            "Adding %(count)d file from '%(directory)s' ...",
-                            "Adding %(count)d files from '%(directory)s' ...",
-                            number_of_files),
-                        mparms,
-                        translate=None,
-                        echo=None
-                    )
-                return (os.path.join(root, f) for f in files)
-
-        def process(result=None, error=None):
-            if result:
-                if error is None:
-                    self.add_files(result)
-                thread.run_task(get_files, process)
-
-        process(True, False)
-
-    def _add_directory_non_recursive(self, path):
+                time.sleep(0.01)
+
+    def _scan_dir(self, ignore_hidden, folders, recursive):
         files = []
-        for f in os.listdir(path):
-            listing = os.path.join(path, f)
-            if os.path.isfile(listing):
-                files.append(listing)
-        number_of_files = len(files)
-        if number_of_files:
-            mparms = {
-                'count': number_of_files,
-                'directory': path,
-            }
-            log.debug("Adding %(count)d files from '%(directory)r'" %
-                      mparms)
-            self.window.set_statusbar_message(
-                ngettext(
-                    "Adding %(count)d file from '%(directory)s' ...",
-                    "Adding %(count)d files from '%(directory)s' ...",
-                    number_of_files),
-                mparms,
-                translate=None,
-                echo=None
-            )
-            # Function call only if files exist
+        local_folders = list(folders)
+        while local_folders:
+            current_folder = local_folders.pop(0)
+            current_folder_files = []
+            for entry in os.scandir(current_folder):
+                if ignore_hidden and is_hidden(entry.path):
+                    continue
+                if recursive and entry.is_dir():
+                    local_folders.extend([entry.path])
```

Comment:
Since you add only one element, `list.append()` is more appropriate.

```python
    local_folders.append(entry.path)
else:
    current_folder_files.append(entry.path)
```

Reply:
Ok.